### PR TITLE
Return a 400 when /decide called with invalid token

### PR DESCRIPTION
**Pull request by macobo**
Sentry: https://sentry.io/organizations/posthog/issues/2133194501/?project=1899813&query=is%3Aunassigned+is%3Aunresolved&statsPeriod=14d

The error messages are now in sync with /capture endpoints as well.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by yakkomajuri** [comment]
flyby potential nit: maybe a 401 is more appropriate given that this is an authentication error?

File: posthog/api/decide.py [hunk 1]
```diff
@@ -92,10 +92,23 @@ def get_decide(request: HttpRequest):
...
         if team is None and token:
             project_id = _get_project_id(data, request)
+
+            if not project_id:
+                return cors_response(
+                    request,
+                    JsonResponse(
+                        {
+                            "code": "validation",
+                            "message": "Project API key invalid. You can find your project API key in PostHog project settings.",
```
Comment (yakkomajuri) [inline posthog/api/decide.py:102]: isn't this checking for `project_id`?

File: posthog/api/decide.py [hunk 2]
```diff
@@ -92,10 +92,23 @@ def get_decide(request: HttpRequest):
...
+                        },
+                        status=400,
+                    ),
+                )
+
             user = User.objects.get_from_personal_api_key(token)
             if user is None:
                 return cors_response(
-                    request, JsonResponse({"code": "validation", "message": "Invalid personal API key.",}, status=400,),
+                    request, JsonResponse({"code": "validation", "message": "Personal API key invalid.",}, status=400,),
```
Comment (yakkomajuri) [inline posthog/api/decide.py:111]: very much a nit: maybe a 401 is more appropriate given that this is an authentication error?

Comment (macobo) [reply posthog/api/decide.py:111]: I was going for consistency with /capture endpoint but you're right!

**Review by yakkomajuri — APPROVED** [review]

**Pull request merged**